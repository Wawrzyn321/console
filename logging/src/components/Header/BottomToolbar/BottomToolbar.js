import React from 'react';
import PropTypes from 'prop-types';

import { Icon } from 'fundamental-react';

BottomToolbar.propTypes = {
  sortDirection: PropTypes.oneOf(['ascending', 'descending']).isRequired,
  updateFilteringState: PropTypes.func.isRequired,
};

export default function BottomToolbar({ sortDirection, updateFilteringState }) {
  function switchSortDirection() {
    updateFilteringState({
      sortDirection: sortDirection === 'ascending' ? 'descending' : 'ascending',
    });
  }

  return (
    <div>
      <span className="link-button fd-has-type-minus-1 fd-has-margin-right-small">
        <Icon glyph="sort" className="fd-has-margin-right-tiny" size="s" />
        last hour
      </span>
      <span
        className="link-button fd-has-type-minus-1"
        onClick={() => switchSortDirection()}
      >
        <Icon glyph="past" size="s" /> {sortDirection}
      </span>
    </div>
  );
}
