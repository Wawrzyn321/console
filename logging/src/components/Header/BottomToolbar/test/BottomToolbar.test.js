import React from 'react';
import renderer from 'react-test-renderer';
import BottomToolbar from './../BottomToolbar';

describe('ApiDetails', () => {
  it('Renders with minimal props', () => {
    const component = renderer.create(
      <BottomToolbar
        updateFilteringState={() => {}}
        sortDirection="ascending"
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
