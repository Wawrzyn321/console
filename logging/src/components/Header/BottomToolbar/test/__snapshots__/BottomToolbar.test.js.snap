// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiDetails Renders with minimal props 1`] = `
<div>
  <span
    className="link-button fd-has-margin-right-small"
  >
    <span
      className="sap-icon--sort sap-icon--s fd-has-margin-right-tiny "
    />
    last hour
  </span>
  <span
    className="link-button"
    onClick={[Function]}
  >
    <span
      className="sap-icon--past sap-icon--s"
    />
     
    ascending
  </span>
</div>
`;
